### DISCUSSION TITLE
Is mitmproxy safe? Why is it marked as malious?

### DISCUSSION THREAD

**Discussion by Idlefase** [mitmproxy/mitmproxy#7283] [Q&A]
Recently I've discovered this project and was willing to install it through homebrew, because I'm skeptical about installing homebrew formulas I wanted to check if it was safe to run by uploading the .app to virustotal. 

To my surprise it was flagged by many anti virus vendors.

Can anyone explain what might have raised this alerts? I know that Avast sees it as a tool or PUP but the other definitions are unclear to me.

Your help is greatly appreciated. 

My excuses if it is a problem that the .app was uploaded to virustotal.

**Comment by mhils** [comment]
These are almost certainly false positives and mitmproxy is safe to use.

We use PyInstaller to package mitmproxy. PyInstaller unfortunately is also used by many malware authors, and thus the PyInstaller wrapper often ends up in AV signatures. This incorrectly flags all PyInstaller-generated binaries, including benign ones.

FWIW, we provide sigstore attestations for all releases. These can be used to verify that the binaries have been built in CI and not been tampered with. So anyone can audit the mitmproxy sources and then also the process that turns them into binaries, which happens fully automated in CI.
